src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MeterListener.cs

```diff
@@ -117,37 +117,41 @@ public void EnableMeasurementEvents(Instrument instrument, object? state = null)
             }
-            else if (measurementCallback is MeasurementCallback<int> intCallback)
+            if (typeof(T) == typeof(byte))
+            {
+                _byteMeasurementCallback = (MeasurementCallback<byte>)(object)measurementCallback;
+            }
+            else if (typeof(T) == typeof(int))
             {
-                _intMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : intCallback;
+                _intMeasurementCallback = (MeasurementCallback<int>)(object)measurementCallback;
             }
-            else if (measurementCallback is MeasurementCallback<float> floatCallback)
+            else if (typeof(T) == typeof(float))
             {
-                _floatMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : floatCallback;
+                _floatMeasurementCallback = (MeasurementCallback<float>)(object)measurementCallback;
             }
-            else if (measurementCallback is MeasurementCallback<double> doubleCallback)
+            else if (typeof(T) == typeof(double))
             {
-                _doubleMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : doubleCallback;
+                _doubleMeasurementCallback = (MeasurementCallback<double>)(object)measurementCallback;
             }
-            else if (measurementCallback is MeasurementCallback<decimal> decimalCallback)
+            else if (typeof(T) == typeof(decimal))
             {
-                _decimalMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : decimalCallback;
+                _decimalMeasurementCallback = (MeasurementCallback<decimal>)(object)measurementCallback;
             }
-            else if (measurementCallback is MeasurementCallback<short> shortCallback)
+            else if (typeof(T) == typeof(short))
             {
-                _shortMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : shortCallback;
+                _shortMeasurementCallback = (MeasurementCallback<short>)(object)measurementCallback;
             }
-            else if (measurementCallback is MeasurementCallback<long> longCallback)
+            else if (typeof(T) == typeof(long))
             {
-                _longMeasurementCallback = (measurementCallback is null) ? ((instrument, measurement, tags, state) => { /* no-op */}) : longCallback;
+                _longMeasurementCallback = (MeasurementCallback<long>)(object)measurementCallback;
             }
             else
             {
-                throw new InvalidOperationException(SR.Format(SR.UnsupportedType, typeof(T)));
+                 throw new InvalidOperationException(SR.Format(SR.UnsupportedType, typeof(T)));
```

Comment:
nit: extra space is prefixed to this line.

Reply:
I fix it